### DISCUSSION TITLE
FAQ: add "why not Tcl"

### DISCUSSION THREAD

**Discussion by andrewshadura** [kdl-org/kdl#273] [General]
For completeness sake. I do see some of the benefits of KDL syntax over a Tcl-based DSL, but it’d be great to hear the "official" view on this 🙂

**Comment by tabatkins** [comment]
I'm not sure I understand. Why *yes* Tcl?

**Reply by andrewshadura** [reply]
What do you mean? You didn’t even consider it in your prior art research? If so, very surprising, given the similarities.

**Reply by tabatkins** [reply]
I didn't design KDL, so no. ^_^ But I know it was based on [SDLang](https://sdlang.org/), basically just making the lang more consistent and extensible.

But still - as someone with no connection to the C community, I just don't know what benefits Tcl has over anything else. From a cursory read over Wikipedia it seems like it is fairly similar grammatically. Is there more to it? What sort of things is it commonly used for?

**Comment by zkat** [comment]
Tcl and KDL are very very different beasts. Tcl is a whole programming language. KDL is a very restricted, config-oriented data language. They have entirely different purposes and use cases and I'm not really sure it's worth the time to compare them.

**Reply by andrewshadura** [reply]
I didn’t mean Tcl as a whole but Tcl’s syntax, of course.
